src/wallet/rpcwallet.cpp

```diff
@@ -2298,6 +2298,13 @@ UniValue walletpassphrase(const JSONRPCRequest& request)
     // Alternately, find a way to make request.params[0] mlock()'d to begin with.
     strWalletPass = request.params[0].get_str().c_str();
 
+    // Get the timeout
+    int nSleepTime = request.params[1].get_int();
+    // Timeout cannot be negative, otherwise it will relock immediately
+    if (nSleepTime < 0) {
```

Comment:
Do we want to allow zero?

Reply:
What would be the point of allowing 0?

Reply:
Perhaps to just test if the passphrase is correct. (You're *currently* allowing zero here.)

Reply:
> (You're currently allowing zero here.)

Right.

It doesn't matter either way IMO.

Reply:
Accepting 2147483647 (years long timeout) does not make too much sense in the time of Meltdown and Spectre vulnerabilities.
Anyway having timeout on the level of single days can be too long for these type of attacks.